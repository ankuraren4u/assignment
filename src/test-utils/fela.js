import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createRenderer } from "fela";
import { RendererProvider, ThemeProvider } from "react-fela";

import { theme } from "./../theme";
Enzyme.configure({ adapter: new Adapter() });

const renderer = createRenderer();

export const felaShallow = node =>
  shallow(
    <RendererProvider renderer={renderer}>
      <ThemeProvider theme={theme}>{node}</ThemeProvider>
    </RendererProvider>
  );

export const felaMount = node =>
  mount(
    <RendererProvider renderer={renderer}>
      <ThemeProvider theme={theme}>{node}</ThemeProvider>
    </RendererProvider>
  );
