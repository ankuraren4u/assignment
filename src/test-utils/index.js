export { felaShallow, felaMount } from "./fela";
