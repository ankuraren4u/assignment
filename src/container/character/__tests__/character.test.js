import { Character } from "..";
import { act, renderHook } from "@testing-library/react-hooks";

import { felaShallow } from "./../../../test-utils/";
import { MemoryRouter, Route, useHistory } from "react-router-dom";
import fetchMock from "jest-fetch-mock";

fetchMock.enableMocks();

const mockBody = [
  {
    character: { id: 1, name: "test1" }
  },
  {
    character: { id: 2, name: "test2" }
  }
];

describe("<Character />", () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  describe("By default", () => {
    let component;
    beforeAll(() => {
      fetch.mockResponse(JSON.stringify(mockBody));
      component = felaShallow(
        <MemoryRouter initialEntries={[`/character/${mockBody[0].character.id}`]}>
          <Route path="/character/:id">
            <Character />
          </Route>
        </MemoryRouter>
      );
    });

    it("renders", () => {
      expect(component.html()).toMatchSnapshot();
    });

    // it("should render the fetched result", () => {
    //   act(() => {
    //     expect(fetch).toHaveBeenCalled();
    //   })
    // });

    // it("should render the fetched result", () => {
    //   act(() => {
    //     const history = useHistory();
    //     history.push(`/character/${mockBody[1].character.id}`);
    //   });

    //   // act(() => {
    //   //   expect(fetch).toHaveBeenCalled();
    //   // })
    // });
  });

  
});
