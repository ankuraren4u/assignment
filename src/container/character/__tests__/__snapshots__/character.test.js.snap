// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Character /> By default renders 1`] = `"<div class=\\"ui-container a b c\\"><div class=\\"table-view b d e f g\\"></div></div>"`;
